src/components/PDFView/index.ios.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import {KeyboardAvoidingView} from 'react-native';
```

Comment:
But this works without changing them.

Reply:
No it doesn't. Note that the form is no longer vertically centered.

<img width="348" alt="kav_1" src="https://user-images.githubusercontent.com/11221215/182221660-e7698ca6-8234-4916-a44c-6ebf462b46ce.png">

Reply:
Did you try with other JSX suggestions?

Reply:
What JSX suggestions? Can you please provide specific direction?

Reply:
Oh sorry, do you mean remove the `View` before the `ScrollView` ?

Reply:
There are other suggestions below, I was referring to them.

Reply:
Have you tested these changes on all platforms?

Reply:
I tested web and iOS. Is it failing somewhere? I am sure that can be fixed. 

Mostly the motive is to get away from https://github.com/Expensify/App/pull/9141#discussion_r934772613. We don't need it. The existing padding-based approach should work fine unless I am missing something.

Reply:
Ok, I'll give it a try.

Reply:
For me the JSX changes break web. The form is no longer vertically centered. Can you send me your `PDFPasswordForm.js`? Did you maybe also make changes to `index.js`?

<img width="903" alt="scroll_view_refactor" src="https://user-images.githubusercontent.com/11221215/182241016-17ee2439-f2a7-4723-80f9-4965880fbb2c.png">

Reply:
I had a typo here https://github.com/Expensify/App/pull/9141#discussion_r933383166 `contentContainerStyles` instead of `contentContainerStyle`. Check all the suggestions and use the correct prop or syntax. There could be typos as such.

Reply:
Now the form is totally broken on iOS.

Apparently everything is working for you so why not send me the code? It'd be even better if you would push your changes to a branch.

<img width="349" alt="scroll_view_refactor_2" src="https://user-images.githubusercontent.com/11221215/182289371-173d6fc8-9f56-4824-83ac-67803d140c47.png">

Reply:
I already discarded all the changes. You can try to look for the issue and adjust the styles.

Reply:
I already spent many hours on this code to get everything to render properly on all platforms. The code works correctly and, as far as I can tell, your proposed changes do not work.

If you can't provide a working solution then we'll need to increase the price of the project and add a milestone for this task.

How would you like to proceed?

Reply:
You can use this patch. 

```diff
diff --git a/src/components/PDFView/BasePDFViewNative.js b/src/components/PDFView/BasePDFViewNative.js
index c4aa64003..4df86d84e 100644
--- a/src/components/PDFView/BasePDFViewNative.js
+++ b/src/components/PDFView/BasePDFViewNative.js
@@ -136,7 +136,7 @@ class BasePDFViewNative extends Component {
             ? styles.w100 : {};
 
         return (
-            <View style={containerStyles}>
+            <View style={[containerStyles, styles.flex1]}>
                 {this.state.shouldAttemptPdfLoad && (
                     <TouchableWithoutFeedback style={touchableStyles}>
                         <PDF
diff --git a/src/components/PDFView/PDFPasswordForm.js b/src/components/PDFView/PDFPasswordForm.js
index 66c95b5cc..4e3752042 100644
--- a/src/components/PDFView/PDFPasswordForm.js
+++ b/src/components/PDFView/PDFPasswordForm.js
@@ -87,51 +87,55 @@ class PDFPasswordForm extends Component {
 
     render() {
         const containerStyle = this.props.isSmallScreenWidth
-            ? [styles.p5, styles.w100]
+            ? [styles.flex1, styles.w100]
             : styles.pdfPasswordForm.wideScreenWidth;
 
         return (
             <>
                 {this.state.shouldShowForm ? (
-                    <View style={containerStyle}>
-                        <ScrollView keyboardShouldPersistTaps="handled">
-                            <Text style={styles.mb4}>
-                                {this.props.translate('attachmentView.pdfPasswordForm.formLabel')}
-                            </Text>
-                            <TextInput
-                                label={this.props.translate('common.password')}
-                                autoComplete="off"
-                                autoCorrect={false}
-                                textContentType="password"
-                                onChangeText={this.updatePassword}
-                                returnKeyType="done"
-                                onSubmitEditing={this.submitPassword}
-                                errorText={this.state.validationErrorText}
-                                onBlur={this.validate}
-                                autoFocus={this.props.shouldAutofocusPasswordField}
-                                secureTextEntry
-                            />
-                            {this.props.isPasswordInvalid && (
-                                <View style={[styles.flexRow, styles.alignItemsCenter, styles.mt3]}>
-                                    <Icon src={Expensicons.Exclamation} fill={colors.red} />
-                                    <View style={[styles.flexRow, styles.ml2, styles.flexWrap, styles.flex1]}>
-                                        <Text style={styles.mutedTextLabel}>
-                                            {this.props.translate('attachmentView.passwordIncorrect')}
-                                        </Text>
-                                    </View>
-                                </View>
-                            )}
-                            <Button
-                                text={this.props.translate('common.confirm')}
-                                onPress={this.submitPassword}
-                                style={styles.pt4}
-                                isLoading={this.props.shouldShowLoadingIndicator}
-                                pressOnEnter
-                            />
-                        </ScrollView>
-                    </View>
+                    <ScrollView
+                        keyboardShouldPersistTaps="handled"
+                        style={containerStyle}
+                        contentContainerStyle={[styles.ph5, styles.flex1, styles.justifyContentCenter]}
+                    >
+                        <Text style={styles.mb4}>
+                            {this.props.translate('attachmentView.pdfPasswordForm.formLabel')}
+                        </Text>
+                        <TextInput
+                            label={this.props.translate('common.password')}
+                            autoComplete="off"
+                            autoCorrect={false}
+                            textContentType="password"
+                            onChangeText={this.updatePassword}
+                            returnKeyType="done"
+                            onSubmitEditing={this.submitPassword}
+                            errorText={this.state.validationErrorText}
+                            onBlur={this.validate}
+                            autoFocus={this.props.shouldAutofocusPasswordField}
+                            secureTextEntry
+                        />
+                        {this.props.isPasswordInvalid && (
+                        <View style={[styles.flexRow, styles.alignItemsCenter, styles.mt3]}>
+                            <Icon src={Expensicons.Exclamation} fill={colors.red} />
+                            <View style={[styles.flexRow, styles.ml2, styles.flexWrap, styles.flex1]}>
+                                <Text style={styles.mutedTextLabel}>
+                                    {this.props.translate('attachmentView.passwordIncorrect')}
+                                </Text>
+                            </View>
+                        </View>
+                        )}
+                        <Button
+                            text={this.props.translate('common.confirm')}
+                            onPress={this.submitPassword}
+                            style={styles.pt4}
+                            isLoading={this.props.shouldShowLoadingIndicator}
+                            pressOnEnter
+                        />
+                    </ScrollView>
                 ) : (
-                    <PDFInfoMessage onShowForm={this.showForm} />
+                    <View style={[styles.flex1, styles.justifyContentCenter]}>
+                        <PDFInfoMessage onShowForm={this.showForm} />
+                    </View>
                 )}
             </>
         );
diff --git a/src/components/PDFView/index.ios.js b/src/components/PDFView/index.ios.js
index bac71b6b2..9ea9be0d3 100644
--- a/src/components/PDFView/index.ios.js
+++ b/src/components/PDFView/index.ios.js
@@ -1,36 +1,18 @@
 import React from 'react';
-import {KeyboardAvoidingView} from 'react-native';
-import CONST from '../../CONST';
+import KeyboardAvoidingView from '../KeyboardAvoidingView';
 import BasePDFViewNative from './BasePDFViewNative';
 import * as pdfViewPropTypes from './pdfViewPropTypes';
 import withWindowDimensions from '../withWindowDimensions';
 import styles from '../../styles/styles';
 
-const PDFView = (props) => {
-    const keyboardVerticalOffset = props.isSmallScreenWidth
-        ? CONST.PDF_PASSWORD_FORM.KEYBOARD_VERTICAL_OFFSET.SMALL_SCREEN
-        : CONST.PDF_PASSWORD_FORM.KEYBOARD_VERTICAL_OFFSET.DEFAULT;
-
-    const keyboardAvoidingViewStyles = [
-        styles.alignItemsCenter,
-        styles.justifyContentCenter,
-        styles.w100,
-    ];
-
-    return (
-        <KeyboardAvoidingView
-            style={keyboardAvoidingViewStyles}
-            contentContainerStyle={keyboardAvoidingViewStyles}
-            behavior="position"
-            keyboardVerticalOffset={keyboardVerticalOffset}
-        >
-            <BasePDFViewNative
-                sourceURL={props.sourceURL}
-                style={[props.style, styles.w100]}
-            />
-        </KeyboardAvoidingView>
-    );
-};
+const PDFView = props => (
+    <KeyboardAvoidingView>
+        <BasePDFViewNative
+            sourceURL={props.sourceURL}
+            style={[props.style, styles.w100]}
+        />
+    </KeyboardAvoidingView>
+);
 
 PDFView.propTypes = pdfViewPropTypes.propTypes;
 PDFView.defaultProps = pdfViewPropTypes.defaultProps;
diff --git a/src/components/PDFView/index.js b/src/components/PDFView/index.js
index 48f97c6c9..6ca8673dc 100644
--- a/src/components/PDFView/index.js
+++ b/src/components/PDFView/index.js
@@ -104,7 +104,6 @@ class PDFView extends Component {
         const outerContainerStyle = [
             styles.w100,
             styles.h100,
-            styles.dFlex,
             styles.justifyContentCenter,
             styles.alignItemsCenter,
         ];
```

Reply:
Thanks for the code, @parasharrajat.